### PR TITLE
SCHED-666: Pickling OCS Resource and Env service now works.

### PR DESCRIPTION
**Pull request by sraaphorst**
This code allows the OCS Resource and Env services to be pickled to increase the speed of the executing code.
The bulk of the speed increase is needed in the viscalcs, but this was a good test that `Resource` is now stable and working correctly.

The `Sources` do not load twice now as they did before, and logging is added to display whether the services are loaded from pickles or created.

Plans based on pickled services or freshly created services are the same, as expected.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -134,5 +134,5 @@ plans.lock
 # Edit: include this file now as access to it online requires authorization.
 #scheduler/services/resource/data/telescope_schedules.xlsx
 
-
-
+# Pickled services
+/pickles
```
Comment (stroncod) [inline .gitignore:138]: If there are never committed to the repo, at the moment of creating the Docker image the pickles are going to be absent and then in Heroku would basically create everything from scratch, rendering this pointless.

**Review by stroncod — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (sraaphorst) [reply .gitignore]: Fair enough. I adjusted the file.

**Comment by sraaphorst** [comment]
On second thought, I disagree with the assessment. We include many files that will never be added to the repo, and it's good practice to do so:

https://github.com/gemini-hlsw/scheduler/blob/main/.gitignore

Ensuring that there can never be an accidental addition of the pickle files is important, because if they are added, they may cause problems.

Screenshot 2024-03-27 at 12 53 04 AM

**Comment by sraaphorst** [comment]
Actually, never mind the previous comment: upon examination, we have `*.pickle` already in `.gitignore`, so this should be irrelevant.

**Review by stroncod — APPROVED** [review]

**Pull request merged**